projects/rulesets-base/src/rule-runner/data-constructors.ts

```diff
@@ -46,6 +46,35 @@ const createFactsWithRaw = <T extends OpenApiKind>(
 
   return factsWithRaw;
 };
+const createPropertyFactsWithRaw = <T extends OpenApiKind>(
+  nodeDetailMap: Map<string, NodeDetail<T>>,
+  key: SpecFactsFrom,
+  openApiSpec: OpenAPIV3.Document
+): Map<
+  string,
+  FactVariant<T> & { raw: any; parent: OpenAPIV3.NonArraySchemaObject }
+> => {
+  const factsWithRaw: Map<
+    string,
+    FactVariant<T> & { raw: any; parent: OpenAPIV3.NonArraySchemaObject }
+  > = new Map();
+
+  for (const [name, nodeDetail] of nodeDetailMap.entries()) {
+    const nodeFact = nodeDetail[key];
+    if (nodeFact) {
+      const parentPath = jsonPointerHelpers.pop(
+        jsonPointerHelpers.pop(nodeFact.location.jsonPath)
+      );
+      factsWithRaw.set(name, {
+        ...nodeFact,
+        raw: jsonPointerHelpers.get(openApiSpec, nodeFact.location.jsonPath),
+        parent: jsonPointerHelpers.get(openApiSpec, parentPath),
```

Comment:
This is just a pointer so we're not dup'ing this in memory. It looks very verbose in the snapshots though